### DISCUSSION TITLE
分享在proxmox VE8.0下的PaoPaoDNS和PaoPaoGateWay的网络配置

### DISCUSSION THREAD

**Discussion by hsj1992** [kkkgo/PaoPaoGateWay#23] [配置分享]
一：主路由方面：

- 主路由硬件：`TP-LINK ER2260T `
- 主路由IP地址： `192.168.31.1`
- 主路由上的设置：
  - 1.连接桥接光猫进行拨号
  - 2.作为DHCP服务器，设置DNS服务器为PaoPaoDNS的地址`192.168.31.4`
    - （路由器左侧菜单”基本设置“-”DHCP服务“-”DHCP服务“）
    - 对服务接口`LAN`的DHCP服务器进行编辑，`首选DNS服务器`填写`192.168.31.4`
  - 3.设置静态路由
    - （路由器左侧菜单”传输控制“-”路由设置“-”静态路由“）
    - 规则名称：PaoPaoGateWay
    - 目的地址：`11.0.0.0`
    - 子网掩码：`255.0.0.0`
    - 出接口：`LAN`
    - 下一跳：`192.168.31.5`
    - （设置telegram的静态路由也差不多是这样操作）

二：Proxmox 虚拟环境方面：

- 硬件：`倍控 N5105 四网口主机`
- 管理IP地址：`192.168.31.2`
- Proxmox上的设置：
    - 运行一个Docker LXC容器（我直接使用的[tteck的script](https://github.com/tteck/Proxmox)安装的）
      - 运行PaoPaoDNS
      - 编译Clash-Meta核心
      - 做PaoPaoGateWay镜像定制
    - 用`local(pve)`-`ISO镜像`-`从URL下载`功能，通过PaoPaoDNS的http静态文件服务器把PaoPaoGateWay定制镜像iso下载到Proxmox里
    - 运行一个虚拟机
      - 挂载运行PaoPaoGateWay

三：Docker LXC容器（PaoPaoDNS)方面：
- IP地址：`192.168.31.4/24`
- DNS服务器：`192.168.31.1`
- 网关IP地址：`192.168.31.1`
- PaoPaoDNS设置：
  - 基本参照PaoPaoDNS和PaoPaoGateWay的说明
  - 开启5304/7889端口映射
  - bridge模式
  - 使用的portainer，修改环境变量还是很简单的

 `<html>
<body>

name|value 
-- | --
ADDINFO | yes
AUTO_FORWARD | yes
AUTO_FORWARD_CHECK | yes
CN_TRACKER | yes
CNAUTO | yes
CNFALL | yes
CUSTOM_FORWARD | 192.168.31.5:53
DNS_SERVERNAME | PaoPaoDNS,blog.03k.org
DNSPORT | 53
HTTP_FILE | yes
IPV6 | no
PATH | /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
QUERY_TIME | 3000ms
RULES_TTL | 0
SERVER_IP | 192.168.31.4
SOCKS5 | 192.168.31.5:1080
TZ | Asia/Shanghai
UPDATE | weekly
USE_HOSTS | no
USE_MARK_DATA | yes

</body>
</html>

四、PaoPaoGateWay方面：

- network.ini:
```
ip=192.168.31.5
mask=255.255.255.0
gw=192.168.31.1

```
- ppgw.ini:

```
#paopao-gateway

# mode=socks5|ovpn|yaml|suburl|free
# default: socks5
mode=yaml

# Set fakeip's CIDR here
# default: fake_cidr=7.0.0.0/8
fake_cidr=11.0.0.0/8

# Set your trusted DNS here
# default: dns_ip=1.0.0.1
dns_ip=192.168.31.4
# default: dns_port=53
# If used with PaoPaoDNS, you can set the 5304 port
dns_port=5304

# Clash's web dashboard
clash_web_port="80"
clash_web_password="clashpass"

# default:openport=no
# socks+http mixed 1080
openport=yes

# default: udp_enable=no
udp_enable=yes

# default:30
sleeptime=30

# socks5 mode settting
# default: socks5_ip=gatewayIP
socks5_ip="192.168.31.1"
# default: socks5_port="7890"
socks5_port="7890"

# ovpn mode settting
# The ovpn file in the same directory as the ppgw.ini.
# default: ovpnfile=custom.ovpn
ovpnfile="custom.ovpn"
ovpn_username=""
ovpn_password=""

# yaml mode settting
# The yaml file in the same directory as the ppgw.ini.
# default: yamlfile=custom.yaml
yamlfile="custom.yaml"

# suburl mode settting
suburl=""
subtime=1d

# fast_node=check/yes/no
fast_node=yes
test_node_url="https://cp.cloudflare.com"
ext_node="Traffic|Expire| GB|Days|Date"
cpudelay="3000"

# dns burn setting
# depand on fast_node=yes & mode=suburl/yaml
dns_burn=yes
# If used with PaoPaoDNS, you can set the PaoPaoDNS:53
ex_dns="192.168.31.4:53"
```

**Comment by kkkgo** [comment]
谢谢分享，如果遇到问题欢迎反馈。

**Comment by biggbuddy** [comment]
请教一下, 为什么 "Docker LXC容器（PaoPaoDNS)" 这里的DNS服务器要设置为192.168.31.1, 也就是主路由的IP?

**Reply by kkkgo** [reply]
Docker宿主的DNS设置成什么其实无所谓，只要你保证你能正常把docker镜像拉下来就行。

**Reply by biggbuddy** [reply]
我的理解, 这应该是已经完全配置好后的网络状态, 意思是当全部运行起来之后, 这里PaoPaoDNS LXC容器的DNS服务器配置已经没有意义了吗? 配什么都可以?

**Reply by kkkgo** [reply]
你的DNS能拉镜像就行，毕竟你宿主可能还会跑别的docker，或者更新镜像。设置成ppgw的DNS的话能全局代理加速拉镜像。
要注意如果docker的宿主也设置成PaoPaoDNS的话，容器没起来的话那你宿主就没DNS用了。

**Comment by fsiisx** [comment]
已经按照这篇教程搭建起来了，运行正常，感谢大佬的付出。
这里我有1个疑问，烦请大佬解答一下。
PaoPaoGateWay的dns填啥？填主路由的ip还是PaoPaoGateWay的ip还是PaoPaoDNS的ip？
我看还有dns1和dns2，我目前填的PaoPaoDNS，假如都填且填的PaoPaoDNS，是推荐用两台设备来跑PaoPaoDNS，可以起到一个冗余容灾的作用吗？

**Reply by kkkgo** [reply]
PaoPaoGateWay的网卡dns仅用于拉取ppgw.ini无其他用途。

**Reply by fsiisx** [reply]
> PaoPaoGateWay的网卡dns仅用于拉取ppgw.ini无其他用途。

那节点的域名解析呢？用的PaoPaoDNS的5304端口吗？

**Reply by kkkgo** [reply]
是的用的5304端口。
```shell
# Set your trusted DNS here
# default: dns_ip=1.0.0.1
dns_ip=192.168.31.4
# default: dns_port=53
# If used with PaoPaoDNS, you can set the 5304 port
dns_port=5304
```
如果你使用了dns burn功能指定N个DNS，则会额外使用其他DNS再额外解析N次节点。
```shell
# dns burn setting
# depand on fast_node=yes & mode=suburl/yaml
dns_burn=yes
# If used with PaoPaoDNS, you can set the PaoPaoDNS:53
ex_dns="192.168.31.4:53"
```

**Comment by fsiisx** [comment]
这里更新一点，pve虚拟机运行PaoPaoGateWay，处理器类别需要选择x86-64-v3，默认的是x86-64-v2-AES启动报错。
<img width="861" alt="image" src="https://github.com/kkkgo/PaoPaoGateWay/assets/17349950/513defe9-9715-4fb4-a801-6667d8fa950c">

**Reply by kkkgo** [reply]
取决于你的CPU和内置的clash内核。如果你手动内置了V3架构的内核则需要V3的CPU支持。默认的clash内核是兼容的。

**Reply by Mxzjxs** [reply]
> 取决于你的CPU和内置的clash内核。如果你手动内置了V3架构的内核则需要V3的CPU支持。默认的clash内核是兼容的。

用host呢？或者用哪个更推荐

**Reply by kkkgo** [reply]
参考https://github.com/kkkgo/PaoPaoGateWay/discussions/65#discussioncomment-8577918

**Reply by kkkgo** [reply]
”我应该下载哪一个文件参考“ https://github.com/MetaCubeX/mihomo/wiki/FAQ
V3参考 https://go.dev/wiki/MinimumRequirements
While performance is expected to improve when a higher minimum microarchitecture is requested, this might not be true in all cases. Benchmark your performance-critical code to verify performance improvements.
Setting, for example, GOAMD64=v3, will allow the Go compiler to use AVX2 instructions in the generated binaries (which may improve performance in some cases); but these binaries will not run on older x86 processors that don’t support AVX2.
推荐是如果原有内核已经能满足需求，则没有更换的必要。
如果需要更换内核，请根据你的CPU兼容性选择。在PVE里面，选择host应该意味着使用宿主相同的CPU兼容性。

**Comment by zfkira** [comment]
一个新手小问题，PVE下 LXC 的 CT 模板建议用 alpine 还是 debian 12, 然后进去再 安装 docker. 这部分教程我可以参考哪儿呢？

**Reply by hsj1992** [reply]
我懒，直接用别人写好的脚本：https://tteck.github.io/Proxmox/
有个问题是脚本里一些下载操作是github直连的，所以如果你网络环境直连访问github有点麻烦的话，脚本运行可能会不大顺。

**Reply by biggbuddy** [reply]
官方的建议是PVE下使用docker不要用LXC， 而是用VM

当然了， 我也在用LXC， 也没发现有什么问题。官方也认为大概率不会有问题

**Reply by gavinlin1123** [reply]
才发现有这么个东西，对懒人来说，实在是太太太实用了，谢谢兄弟

**Comment by gavinlin1123** [comment]
> 我懒，直接用别人写好的脚本：https://tteck.github.io/Proxmox/ 有个问题是脚本里一些下载操作是github直连的，所以如果你网络环境直连访问github有点麻烦的话，脚本运行可能会不大顺。
